src/ModalBase/ModalLayer.jsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/*  TODO: The first of these two disabled linters is okay.
+Focus lock is handling the keyboard for us. The second I'm not sure */
+
+import React, { useMemo, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { FocusOn } from 'react-focus-on';
+import Portal from './Portal';
+import ModalContext from './ModalContext';
+
+const overlayStyles = {
+  zIndex: 3050, // $zindex-modal: 1050 !default;
+  overflow: 'auto',
+  position: 'fixed',
+  top: 0,
+  left: 0,
+  width: '100vw',
+  height: '100vh',
+};
+
+const useMemoizedValues = ({ close, isOpen, isBlocking }) => useMemo(() => ({
```

Comment:
Why are you pulling this out of the main component?

Reply:
See below